extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,152 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+export const VALIDATION_OBJECTS = {
+  'aggregateRating': {
+    'ratingValue': [validateRequired, validateNumber],
+    'reviewCount': [validateRequired, validateNumber],
+    'reviewUrl': [validateRequired, validateURLs],
+  },
+  'productValidationConfig': {
+    /* Required Attrs */
+    'productUrl': [validateRequired, validateString],
+    'productId': [validateRequired, validateString],
+    'productTitle': [validateRequired, validateString],
+    'productBrand': [validateRequired, validateString],
+    'productPrice': [validateRequired, validateNumber],
+    'productImages': [validateRequired, validateURLs],
+    'productPriceCurrency': [validateRequired, validateString],
+    'aggregateRating': [validateRequired, validateObject],
```

Comment:
`validateObject` looks a lot like `validateConfig`. I would suggest preserving a similar signature. This way you don't need a shared object, instead a unique one like `aggregateRatingValidation`.

```suggestion
    'aggregateRating': [validateRequired, createValidateConfig(aggregateRatingValidation)],
```

```js
function createValidateConfig(validation) {
  return (field, value) => validateConfig(value, validation);
}
```

Reply:
Sounds good!